src/platform-implementation-js/namespaces/app-menu.ts

```diff
@@ -102,11 +104,85 @@ Items with CollapsiblePanels can also have accessories which provide primary act
 */
 export default class AppMenu {
   #driver;
+  #events = stream<
+    {
+      type: 'collapseToggled';
+      /** Whether or not the AppMenuBurger is open or not. */
+      open: boolean;
+      /**
+       * 'start' is when a `transactionstart` event is fired on a collapsible panel.
+       * 'end' is when a `transactioncancel` event is fired on a collapsible panel. */
+      stage: 'start' | 'end';
+    },
+    unknown
+  >((emitter) => {
+    const f = async () => {
+      const isShown = await this.isShown();
+
+      if (!isShown) {
+        return;
+      }
+
+      const appMenu = await GmailElementGetter.getAppMenuAsync();
+
+      if (!appMenu) {
+        return;
+      }
+
+      const onTransition =
+        (stage: 'start' | 'end') =>
+        async ({ target }: TransitionEvent) => {
+          const { TOGGLE_OPEN_STATE } = CollapsiblePanelView.elementCss;
+
+          if (
+            target instanceof HTMLElement &&
+            target.matches(CollapsiblePanelView.elementSelectors.NATIVE) &&
+            ((stage === 'start' &&
+              target.classList.contains(TOGGLE_OPEN_STATE)) ||
+              // TOGGLE_OPEN_STATE is removed on transitioncancel TransitionEvents
+              stage === 'end')
+          ) {
+            emitter.emit({
+              type: 'collapseToggled',
+              open: this.isMenuCollapsed(),
+              stage,
+            });
+          }
+        };
+
+      appMenu.parentElement!.addEventListener(
+        'transitionstart',
+        onTransition('start')
+      );
+      appMenu.parentElement!.addEventListener(
+        'transitioncancel',
+        onTransition('end')
+      );
+    };
+
+    f();
+  });
 
   constructor(driver: GmailDriver) {
     this.#driver = driver;
   }
 
+  /**
+   * A stream of events related to the AppMenu.
+   *
+   * @note If the AppMenu is not shown, this stream will not emit any events.
+   */
+  get events() {
+    return this.#events;
+  }
+
+  /**
+   * @returns whether or not the AppMenu Burger is collapsed or not.
+   */
+  isMenuCollapsed() {
```

Comment:
Q: Shouldn't this be named "open" based on `isAppBurgerMenuOpen`?

Reply:
I'm ok with either. I feel like we've gone back and for on the name for collapsed or open since #849.

Reply:
Changed to `isMenuOpen`.